v2/util/pbconv.go

```diff
@@ -0,0 +1,80 @@
+package util
+
+import (
+	"errors"
+
+	pbv1 "github.com/libp2p/go-libp2p-circuit/pb"
+	pbv2 "github.com/libp2p/go-libp2p-circuit/v2/pb"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+func PeerToPeerInfoV1(p *pbv1.CircuitRelay_Peer) (peer.AddrInfo, error) {
+	if p == nil {
+		return peer.AddrInfo{}, errors.New("nil peer")
+	}
+
+	id, err := peer.IDFromBytes(p.Id)
+	if err != nil {
+		return peer.AddrInfo{}, err
+	}
+
+	var addrs []ma.Multiaddr
```

Comment:
addrs := make([]ma.Multiaddr, 0 , len(p.Addrs )) for allocs.

Reply:
this is a micro-optimization that will do more harm than good -- this is 0-sized almost certainly and we want to keep that nil around.

Reply:
I will humor you however and add that code just so you are happy :)